File: 0011. Container With Most Water/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l,r = 0, len(height) - 1
        dh = float("-inf")

        while l < r:
            val = min(height[l], height[r]) * (r - l)

            if height[l] > height[r]:
                r -= 1

            elif height[l] < height[r]:
                l += 1

            else:
                l += 1
                r -= 1
            
            dh = max(dh, val)

        return dh
    

File: 0011. Container With Most Water/solution.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    int maxArea(std::vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int temp;
        int maxarea = (r - l) * std::min(height[l], height[r]);

        while (l < r){
            temp = (r - l) * std::min(height[l], height[r]);
            maxarea = (temp > maxarea) ? temp : maxarea;
            if (height[l] > height[r]){
                r--;
            }
            else if (height[l] < height[r]){
                l++;
            }
            else{
                l++;
                r--;
            }

        }

        return maxarea;
    }
};
